src/zeek.bif

```diff
@@ -5576,12 +5576,32 @@ function anonymize_addr%(a: addr, cl: IPAddrAnonymizationClass%): addr
 ##
 ## returns: a JSON formatted string.
 ##
-## .. zeek:see:: fmt cat cat_sep string_cat print_raw
+## .. zeek:see:: fmt cat cat_sep string_cat print_raw from_json
 function to_json%(val: any, only_loggable: bool &default=F, field_escape_pattern: pattern &default=/^_/%): string
 	%{
 	return val->ToJSON(only_loggable, field_escape_pattern);
 	%}
 
+## A function to convert a JSON string into arbitrary Zeek data.
+##
+## json_str: The JSON string.
+##
+## t: Type of zeek data.
+##
+## returns: Zeek data.
+##
+## .. zeek:see:: to_json
+function from_json%(json_str: string, t: any%): any
```

Comment:
> It's a bit difficult to work/ignore such errors as they quickly lead to expression errors. At the same, the "happy-path" interface is nicer to work with compared to having a record with an error flag in-between and we don't have exceptions.

After chatting with @rsmmr a bit, given the general nature of input for a function of `from_json()`, a way to recognize errors would be good. The one example I've found is `type ftp_port`, which is a record with `valid` indicator as bool. I think this could work well.

https://github.com/zeek/zeek/blob/e856e953b89bf2719aaefe5bb38cdfa56837f59d/scripts/base/init-bare.zeek#L360-L364

The following might work out - could see a descriptive error message, but that could be extended later, too. The `valid` flag will give a chance to avoid errors if `v` wasn't fully populated.
```
type from_json_result: record { v: any &optional; valid: bool; }
```

cc @ckreibich - I mentioned the easy-happy-path-only interface the other day, but Robin had suggested for JSON parsing a way to handle errors more explicitly would be good.